Makefile

```diff
@@ -172,9 +172,18 @@ bin/Test$(CONFIGURATION)/$(JAVA_INTEROP_LIB): bin/$(CONFIGURATION)/$(JAVA_INTERO
 run-android: $(ATESTS)
 	(cd src/Android.Interop/Tests; $(MSBUILD) $(MSBUILD_FLAGS) '/t:Install;RunTests' $(if $(FIXTURE),/p:TestFixture=$(FIXTURE)))
 
-run-test-jnimarshal: bin/Test$(CONFIGURATION)/Java.Interop.Export-Tests.dll bin/Test$(CONFIGURATION)/$(JAVA_INTEROP_LIB)
+JRE_DLL_CONFIG_SRC=src/Java.Runtime.Environment/Java.Runtime.Environment.dll.config
+JRE_DLL_CONFIG=bin/$(CONFIGURATION)/Java.Runtime.Environment.dll.config
+
+$(JRE_DLL_CONFIG): src/Java.Runtime.Environment/Java.Runtime.Environment.csproj
+	$(MSBUILD) $(MSBUILD_FLAGS) $<
+
+run-test-jnimarshal: bin/Test$(CONFIGURATION)/Java.Interop.Export-Tests.dll bin/Test$(CONFIGURATION)/$(JAVA_INTEROP_LIB) $(JRE_DLL_CONFIG_SRC) $(JRE_DLL_CONFIG)
+	mv -fv $(JRE_DLL_CONFIG) $(JRE_DLL_CONFIG).bak
+	JVM_PATH=`sed -n -E 's/^.*jvm.dll.*target="(.+)".*$$/\1/p' src/Java.Runtime.Environment/Java.Runtime.Environment.dll.config` \
 	MONO_TRACE_LISTENER=Console.Out \
-	$(RUNTIME) bin/$(CONFIGURATION)/jnimarshalmethod-gen.exe -v -L $(JI_MONO_LIB_PATH)mono/4.5 -L $(JI_MONO_LIB_PATH)mono/4.5/Facades "$<"
+	$(RUNTIME) bin/$(CONFIGURATION)/jnimarshalmethod-gen.exe -v --jvm "`sed -n -E 's/^.*jvm.dll.*target="(.+)".*$$/\1/p' $(JRE_DLL_CONFIG_SRC)`" -L "$(JI_MONO_LIB_PATH)"mono/4.5 -L "$(JI_MONO_LIB_PATH)"mono/4.5/Facades "$<"
```

Comment:
er, what? Shouldn't this instead be `--jvm "$(JI_JVM_PATH)"`? See `build-tools/scripts/jdk.mk`.

Reply:
I would prefer `-L "$(JI_MONO_LIB_PATH)mono/4.5"` over `-L "$(JI_MONO_LIB_PATH)"mono/4.5"`. (Note where the `"` is.)